### PR TITLE
Support openLdap Auth Tests

### PR DESCRIPTION
**Pull request by soumyalj**

### CODE REVIEW

File: tests/validation/tests/v3_api/test_auth.py [hunk 1]
```diff
@@ -36,6 +40,22 @@
...
+    os.environ.get("RANCHER_OPENLDAP_SERVICE_ACCOUNT_PASSWORD")
+OPENLDAP_USER_SEARCH_BASE = os.environ.get("RANCHER_OPENLDAP_USER_SEARCH_BASE")
+OPENLDAP_AUTH_USER_PASSWORD = \
+    os.environ.get("RANCHER_OPENLDAP_AUTH_USER_PASSWORD")
+
+global PASSWORD
+
+if AUTH_PROVIDER == "activeDirectory":
+    PASSWORD = AUTH_USER_PASSWORD
+else:
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_auth.py:57]: Let's change it to `elif` as there will be a `freeIpa` at some point.

File: tests/validation/tests/v3_api/test_auth.py [hunk 2]
```diff
@@ -506,35 +528,36 @@ def get_tls(certificate):
...
         "userObjectClass": "inetOrgPerson",
         "userSearchAttribute": "uid|sn|givenName",
-        "userSearchBase": USER_SEARCH_BASE,
-        "serviceAccountPassword": SERVICE_ACCOUNT_PASSWORD
+        "userSearchBase": OPENLDAP_USER_SEARCH_BASE,
+        "serviceAccountPassword": OPENLDAP_SERVICE_ACCOUNT_PASSWORD
     }
-
+    print("LDAP CONFIG IS\n")
+    print(ldap_config)
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_auth.py:560]: We don't need these prints here.

File: tests/validation/tests/v3_api/test_auth.py (hunk 1 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_auth.py:57]: We have to introduce the variables for free IPA password if we want to do that. Hence didn't do that.

Comment (izaac) [reply tests/validation/tests/v3_api/test_auth.py:57]: We don't have as `else` is optional but this is a minor thing that can be improved later on.

File: tests/validation/tests/v3_api/test_auth.py (hunk 2 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_auth.py]: Delted

File: tests/validation/tests/v3_api/test_auth.py (hunk 1 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_auth.py]: Done

File: tests/validation/tests/v3_api/test_auth.py [hunk 3]
```diff
@@ -506,35 +527,34 @@ def get_tls(certificate):
 
 
 def enable_openldap(username, token, enable_url=CATTLE_AUTH_ENABLE_URL,
-                    password=AUTH_USER_PASSWORD, nested=False,
+                    password=OPENLDAP_AUTH_USER_PASSWORD, nested=False,
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_auth.py:530]: why the default value here is `OPENLDAP_AUTH_USER_PASSWORD` not `PASSWORD`? 

In all other places `AUTH_USER_PASSWORD` is replaced by `PASSWORD`, and  `PASSWORD` is equal to `OPENLDAP_AUTH_USER_PASSWORD` when AUTH_PROVIDER == "openLdap".

File: tests/validation/tests/v3_api/test_auth.py [hunk 4]
```diff
@@ -36,7 +40,22 @@
...
+    os.environ.get("RANCHER_OPENLDAP_HOSTNAME_OR_IP_ADDRESS")
+OPENLDAP_SERVICE_ACCOUNT_NAME = \
+    os.environ.get("RANCHER_OPENLDAP_SERVICE_ACCOUNT_NAME")
+OPENLDAP_SERVICE_ACCOUNT_PASSWORD = \
+    os.environ.get("RANCHER_OPENLDAP_SERVICE_ACCOUNT_PASSWORD")
+OPENLDAP_USER_SEARCH_BASE = os.environ.get("RANCHER_OPENLDAP_USER_SEARCH_BASE")
+OPENLDAP_AUTH_USER_PASSWORD = \
+    os.environ.get("RANCHER_OPENLDAP_AUTH_USER_PASSWORD")
+
+global PASSWORD
```
[the remark this answers is not in the document]
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_auth.py:53]: Global since PASSWORD is used across all methods.

[the remark this answers is not in the document]
Comment (jiaqiluo) [reply tests/validation/tests/v3_api/test_auth.py:53]: the parameter is already in the global scope, so we don't need the global keyword

[the remark this answers is not in the document]
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_auth.py]: Done

**Comment by sangeethah** [comment]
LGTM

**Pull request merged**